test/catalog/catalog_test.cpp

```diff
@@ -0,0 +1,350 @@
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  delete accessor;
+
+  txn = txn_manager_->BeginTransaction();
+  accessor = catalog_->GetAccessor(txn, db_);
+  ns_oid = accessor->CreateNamespace("test_namespace");
+  EXPECT_EQ(ns_oid, catalog::INVALID_NAMESPACE_OID);  // Should cause a name conflict
+  txn_manager_->Abort(txn);
+  delete accessor;
+
+  // Get an accessor into the database and validate the catalog tables exist
+  // then delete it and verify an invalid OID is now returned for the lookup
+  txn = txn_manager_->BeginTransaction();
+  accessor = catalog_->GetAccessor(txn, db_);
+  EXPECT_NE(accessor, nullptr);
+  VerifyCatalogTables(accessor);  // Check visibility to me
+  ns_oid = accessor->GetNamespaceOid("test_namespace");
+  EXPECT_TRUE(accessor->DropNamespace(ns_oid));
+  ns_oid = accessor->GetNamespaceOid("test_namespace");
+  EXPECT_EQ(ns_oid, catalog::INVALID_NAMESPACE_OID);
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  delete accessor;
+
+  txn = txn_manager_->BeginTransaction();
+  accessor = catalog_->GetAccessor(txn, db_);
+  EXPECT_FALSE(accessor->DropNamespace(ns_oid));
+  txn_manager_->Abort(txn);
+  delete accessor;
+}
+
+/*
+ * Create and delete a user table.
+ */
+// NOLINTNEXTLINE
+TEST_F(CatalogTests, UserTableTest) {
+  auto txn = txn_manager_->BeginTransaction();
+  auto accessor = catalog_->GetAccessor(txn, db_);
+
+  // Create the column definition (no OIDs)
+  std::vector<catalog::Schema::Column> cols;
+  cols.emplace_back("id", type::TypeId::INTEGER, false,
+                    parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+  cols.emplace_back("user_col_1", type::TypeId::INTEGER, false,
+                    parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+  auto tmp_schema = catalog::Schema(cols);
+
+  auto table_oid = accessor->CreateTable(accessor->GetDefaultNamespace(), "test_table", tmp_schema);
+  EXPECT_NE(table_oid, catalog::INVALID_TABLE_OID);
+  VerifyTablePresent(accessor, accessor->GetDefaultNamespace(), "test_table");
+  // Check lookup via search path
+  EXPECT_EQ(table_oid, accessor->GetTableOid("test_table"));
+  EXPECT_EQ(accessor->GetTable(table_oid), nullptr);  // Check that allocation has not happened
+  auto schema = accessor->GetSchema(table_oid);
+
+  // Verify our columns exist
+  EXPECT_NE(schema.GetColumn("id").GetOid(), catalog::INVALID_COLUMN_OID);
+  EXPECT_NE(schema.GetColumn("user_col_1").GetOid(), catalog::INVALID_COLUMN_OID);
+
+  // Verify we can instantiate a storage object with the generated schema
+  auto table = new storage::SqlTable(&block_store_, schema);
+
+  EXPECT_TRUE(accessor->SetTablePointer(table_oid, table));
+  EXPECT_EQ(common::ManagedPointer(table), accessor->GetTable(table_oid));
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
```

Comment:
We don't support any of those yet. :(

Reply:
Can we leave a TODO both in catalog accessor header file so that other people won't use it accidentally; and in the test cpp file so that we can add the related testcases when we have them implemented?